### PR TITLE
add null and never literals to parser

### PR DESCRIPTION
**Pull request by zojeda**
This is an small improvement in order to parse null and never types, tests were also updated

### CODE REVIEW

**Comment by shiwano** [comment]
Nicely Improvement!

File: test/fixtures/typings/integration/type.d.ts [hunk 1]
```diff
@@ -106,6 +106,8 @@ declare module Type {
   type stringLiteralType = "foobar";
   type booleanLiteralType = true;
   type numberLiteralType = 100;
+  type nullLiteralType = null;
+  type neverLiteralType = never;
```
Comment (shiwano) [inline test/fixtures/typings/integration/type.d.ts]: https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts#L2932

TypeScript's `TypeFlags` does not have `NullLiteralType` and `NeverLiteralType`, so I think that it is better to change this names from `*LiteralType` to `*Type`.

Comment (zojeda) [reply test/fixtures/typings/integration/type.d.ts]: ohh right, thanks !

**Comment by shiwano** [comment]
👍 ✨ ⭐️

**Pull request merged**